telemetry_core/src/main/java/com/newrelic/telemetry/logs/Log.java

```diff
@@ -0,0 +1,212 @@
+    result = 31 * result + (attributes != null ? attributes.hashCode() : 0);
+    result = 31 * result + (serviceName != null ? serviceName.hashCode() : 0);
+    result = 31 * result + (logType != null ? logType.hashCode() : 0);
+    result = 31 * result + (level != null ? level.hashCode() : 0);
+    return result;
+  }
+
+  @Override
+  public String toString() {
+    return "Log{"
+        + "timestamp="
+        + timestamp
+        + ", message='"
+        + message
+        + '\''
+        + ", attributes="
+        + attributes
+        + ", serviceName='"
+        + serviceName
+        + '\''
+        + ", logType='"
+        + logType
+        + '\''
+        + ", level='"
+        + level
+        + '\''
+        + '}';
+  }
+
+  /**
+   * A class for holding the variables associated with a Log object and creating a new Log object
+   * with those variables.
+   */
+  public static class LogBuilder {
+    private long timestamp = System.currentTimeMillis();
+    private String message;
+    private Attributes attributes = new Attributes();
+    private String serviceName; // service.name <- goes in attributes
+    private String logType; // logtype <- goes in attributes
+    private String level;
+
+    /** The point in time (ms since UNIX epoch) that the log entry was created. */
+    public LogBuilder timestamp(long timestamp) {
+      this.timestamp = timestamp;
+      return this;
+    }
+
+    /** The log line itself. */
+    public LogBuilder message(String message) {
+      this.message = message;
+      return this;
+    }
+
+    /** Additional attributes associated with the log entry. */
+    public LogBuilder attributes(Attributes attributes) {
+      this.attributes = attributes;
+      return this;
+    }
+
+    /** The name of the service which produced this log entry. */
+    public LogBuilder serviceName(String serviceName) {
+      this.serviceName = serviceName;
+      return this;
+    }
+
+    /** The log level (eg. INFO, DEBUG, etc) for the log entry. */
+    public LogBuilder level(String logLevel) {
+      this.level = logLevel;
+      return this;
+    }
+
+    /** The type of log entry. This can be helpful for querying your log entries in New Relic. */
+    public LogBuilder logType(String type) {
+      this.logType = type;
+      return this;
+    }
+
+    /** Create the new {@link Log} entry. */
+    public Log build() {
+      Utils.verifyNonBlank(message, "A message is required for a log entry");
+      return new Log(this);
+    }
+
+    /**
+     * Will fill in the {@link #message(String)} with the stack trace from the Throwable. This will
+     * also set the {@link #logType(String)} to be "stackTrace".
+     */
+    public LogBuilder stackTrace(Throwable e) {
+      Utils.verifyNonNull(e);
+      ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+      e.printStackTrace(new PrintStream(bytes));
+      this.message = bytes.toString();
+      this.logType = "stackTrace";
```

Comment:
Is this a convention? 

If so, it doesn't align with our other logging error semantics, here: https://github.com/newrelic/newrelic-exporter-specs/tree/master/logging#errors

Reply:
ooh! I totally forgot there were semantics already designed for some of this stuff. Will address!